[Xiao Ao] iP

xa0412 opened the pull request:
### XanChatBot

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

XanChatBot frees your mind of having to remember things you need to do. It's,

- text-based

- easy to learn

- _**SUPER_ FAST** to use

All you need to do is,

1. download it from [here](url).

2. double-click it.

3. add your tasks.
4. let it manage your tasks for you 😉
5. And it is **FREE**!

Features:

- [x] Managing todo

- [x] Managing deadlines

- [x]  Managing events

- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the main method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

xa0412 committed:
Complete Level 0. Rename, Greet, Exit

xa0412 committed:
remove import

xa0412 committed:
Complete Level 1. Echo

xa0412 committed:
Update Class name and while loop

xa0412 committed:
create method for showList and addList, also updated main method, donw with level 2. Add, List

xa0412 committed:
Done with Level3. Mark as done. Add new class Task to help with my main class (Xan) in task marking

xa0412 committed:
Done with Level 4. ToDos, Events, Deadlines. Added subclass todo, deadline and events to help with my main class (Xan)

xa0412 committed:
updated runtest.bat and runtest.sh file to change Duke to Xan. Test using the I/O redirection technique and make amendment (small replacement changer on Deadline.java) base on the test result

xa0412 committed:
Done with Level 5. Handle Errors. Added a XanException class. modifify Xan class to catch exception
Improve on the welcome message and create a function for welcome message to make the code more readerable.

xa0412 committed:
added Level 6. Delete function
update welcome message to show the description of delete function

xa0412 committed:
Done with A-Enums, use Enum to store todo, event, deadline task type
update Exception to display correct message base on different input.
make other change to make the chatbot more useable.

xa0412 committed:
Deadline class: Update minor toString replacement
Event class: Update minor toString replacement
Xan class: Add loadTask function, Add saveTask function, Update addTask to addTaskMessage
Add xan.txt

xa0412 committed:
Merge branch 'branch-Level-7'

xa0412 committed:
Deadline class: Update date from type String to LocalDate
Task class: Update LocalDate and toString method
XanChatBot class: rename Xan to XanChatBot, updatedate from type String to LocalDate

xa0412 committed:
update xan.txt

xa0412 committed:
Merge branch 'branch-Level-8'

xa0412 committed:
Ui class: Add ui add to handle user interface message
TaskManager class: Add taskManger to handle the function of the type
XanChatBot class: Update the class to make it more OOP

xa0412 committed:
Orignize my XanChatBot project into suitable java packages

xa0412 committed:
Merge branch 'add-gradle-support'

xa0412 committed:
TaskManagerTest: Add test for testAddTodoTask() and testAddEventTask()

xa0412 committed:
TaskManagerTest class: Update to get the correct test case
build.gradle: Update settings

xa0412 committed:
Add JavaDoc for all class and function
Update CodeStandard

xa0412 committed:
XanChatBot class: Add condition for search function
TaskManager class:  Add searchTask() and JavaDoc
Update build.gradle after changing Xan package to xan package

xa0412 committed:
Update Ui class JavaDoc

xa0412 committed:
Merge branch 'branch-Level-9'

xa0412 committed:
Update JavaDoc for XanException class

xa0412 committed:
Merge branch 'branch-A-CodingStandard'

darrenchooji commented on src/main/java/xan/TaskManager.java:
```diff
@@ -0,0 +1,318 @@
+import xan.exception.XanException;
+import xan.ui.Ui;
+
+import java.io.*;
```
Perhaps list the imported classes explicitly?

darrenchooji commented on src/main/java/xan/Event.java:
```diff
@@ -0,0 +1,33 @@
+ * A task representing an event with a description, start time, and end time.
+ */
+public class Event extends Task {
+    protected String startTime;
```
Any reasons why you use 'protected' rather than 'private'?

darrenchooji commented on src/test/java/xan/TaskManagerTest.java:
```diff
@@ -0,0 +1,28 @@
+package xan;
+
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.*;
```
Perhaps you could list out the imported classes explicitly?

xa0412 committed:
Update build.gradle
Add checkstyle.xml and supressions.xml for GUI
Add DialogBox.java, Launcher.java and MainWindow.java
Add DialogBox.fxml and MainWindow.fxml
Add DaDuke.png and DaUser.png
Update TaskManager class and XanChatBot class to implement GUI

xa0412 committed:
Update build.gradle
Add checkstyle.xml and supressions.xml for GUI
Add DialogBox.java, Launcher.java and MainWindow.java
Add DialogBox.fxml and MainWindow.fxml
Add DaDuke.png and DaUser.png
Update TaskManager class and XanChatBot class to implement GUI

xa0412 committed:
Merge branch 'branch-Level-10'

xa0412 committed:
Update TaskManagerTest.java
Update Event.java, change field from protected to private

xa0412 committed:
Update Task class: change fields to privates
Add A-checkStyle tag

Siyan-G reviewed (commented):
Overall good jobz on java coding standards!

Siyan-G commented on src/main/java/task/TaskManager.java:
```diff
@@ -0,0 +1,339 @@
+ * to display tasks to the user and supports different types of tasks such as Todo, Deadline, and Event.
+ */
+public class TaskManager {
+    private static final ArrayList<Task> listArray = new ArrayList<>();
```
According to java coding standard, you should use screaming-snake characters for naming the constant variables. Love the way you declare the list too!

Siyan-G commented on src/main/java/task/TaskManager.java:
```diff
@@ -0,0 +1,339 @@
+                String taskDetails = line.substring(7);
+                Task task = null;
+
+                switch (taskTypeChar) {
+                case 'T':
+                    task = new Todo(taskDetails);
+                    break;
+                case 'D':
+                    if (taskDetails.contains("(by: ")) {
+                        String[] parts = taskDetails.split("\\(by: ");
+                        String description = parts[0].trim();
+                        LocalDate date = LocalDate.parse(parts[1].replace(")", "").trim());
+                        task = new Deadline(description, date);
+                    }
+                    break;
+                case 'E':
+                    if (taskDetails.contains("(from: ") && taskDetails.contains("to: ")) {
+                        String[] parts = taskDetails.split("(\\(from: )| (to: )");
+                        String description = parts[0].trim();
+                        String startTime = parts[1].trim();
+                        String endTime = parts[2].replace(")", "").trim();
+                        task = new Event(description, startTime, endTime);
+                    }
+                    break;
+                default:
+                    throw new IllegalArgumentException("Unknown task type: " + taskTypeChar);
+                }
```
Good practice of coding standard for switch cases without indentation !

Siyan-G commented on src/main/java/task/TaskManager.java:
```diff
@@ -0,0 +1,339 @@
+     *
+     * @return A string containing the list of tasks or a message if empty.
+     */
+    public String showList() {
+        if (listArray.isEmpty()) {
+            return "No tasks found";
+        } else {
+            StringBuilder result = new StringBuilder();
+            result.append("Here are the tasks in your list:\n");
+            for (int i = 0; i < listArray.size(); i++) {
+                Task task = listArray.get(i);
+                result.append((i + 1)).append(". ").append(task.toString()).append("\n");
+            }
+            return result.toString();
+        }
+    }
```
Perfect example of how indentations are used for loops!

Siyan-G commented on src/main/java/task/TaskManager.java:
```diff
@@ -0,0 +1,339 @@
+        return listArray.get(index);
+    }
+
+    /**
+     * Searches for tasks in the list that contain the given keyword in their description.
+     * If no keyword is provided or the keyword is empty, prompts the user to provide a valid keyword.
+     * Displays all matching tasks or a message if no tasks are found.
+     *
+     * @param chat The command string containing the search keyword (e.g., "search keyword").
+     */
+    public String searchTask(String chat) {
```
Perfect example on how to add Javadoc to the methods!

xa0412 committed:
test: add assertions to TaskManager.java and XanChatBot.java
This commit introduces assertions in `TaskManager.java` and `XanChatBot.java` to verify key assumptions and ensure the correctness of the code during runtime.

Changes include:
- Added assertions in `TaskManager.java` to verify task validity (e.g., loadTask(), case for Dealline and Event, saveTask()).
- Added assertions in `XanChatBot.java` to validate input parameters and expected behavior during interaction with the chatbot.

These assertions will help identify potential issues early during testing and ensure data integrity within the application.

xa0412 committed:
Merge pull request #2 from xa0412/branch-A-Assertions

test: add assertions to TaskManager.java and XanChatBot.java

xa0412 committed:
Merge branch 'master' of https://github.com/xa0412/ip

xa0412 committed:
Refactor: Improve code quality in TaskManager.java and XanChatBot.java
This commit refactors TaskManager.java and XanChatBot to improve code quality, readability, and maintainability.

Changes include:
-Refactored logic in methods like enum TaskType, addTask(), showTask() and searchTask() to improve clarity and reduce duplication.
-Enhanced method descriptions and added inline comments for better understanding of the code.
-Improved consistency and readability of conditionals

These changes aim to streamline task management, enhance code readability, and ensure that tasks are handled efficiently with minimal redundancy.

xa0412 committed:
Merge branch 'branch-A-CodeQuality'

xa0412 committed:
Update TaskManager.java on codeQuality
-Update constant variables using SCREAMING_SNAKE_CASE

xa0412 committed:
Update TaskManager and XanChatBot class message.

xa0412 committed:
Update User.png and Xan.png to enhance UI consistency and improve user experience

Replace the existing user profile avatar with a more visually appealing, gender-neutral design.
Ensure better alignment with the chatbot’s futuristic theme.
Adjust brightness and styling for a more balanced and user-friendly interface.

xa0412 committed:
Update boolean found field to isFound field to sound like booleans
Update method name loadTask() to loadTaskFromFile() for meaniful naming

xa0412 committed:
Update MainWindow class: giving credit for reused work

xa0412 committed:
Main.java: Add setTitle() to get title of the application window

xa0412 committed:
Update Ui.java to improve GUI

xa0412 committed:
Update Ui class: Correctly discribed the command message to allow user to input commond

xa0412 committed:
TaskMaanger class: Fix bug where application does not compile after getting date from deadline tasks
Ui class: Update command message for better user interface
XanChatBot class: Update error message after input incorrect input
Add Ui.png to set up product website

xa0412 committed:
Merge branch 'DeadlineEdit'

xa0412 committed:
Update Ui.png to the correct name formate

xa0412 committed:
Update README.md to add a user guide for my XanChatBot

xa0412 committed:
Update README.md to the correct path so that image could abe shown

xa0412 committed:
Update TaskManager class: Make LIST_ARRAY be be non-static
Update Ui class: Update welcome message
Update TaskManagerTest class: Test 4 method using JUnit
Update EXPECTED.TXT and input.txt to provide a correct message

xa0412 committed:
Update XanChatBot class by making file path to provide a runtime-accessible path
Update TaskManager class by update saveTask() to expect a runtime-accessible path

xa0412 committed:
Merge branch 'Branch-JarFile'

xa0412 committed:
Update XanChatBot: Remove unuse import

xa0412 committed:
Merge branch 'branch-Improve-UI'

xa0412 committed:
Ui.java: Remove supressWarning

xa0412 committed:
Update Ui.java: Improve java Doc

xa0412 committed:
Update Ui.java: Imporve javaDoc

xa0412 committed:
Update Gui for betterGui

xa0412 committed:
Merge branch 'branch-improve-GUI'
